.github/workflows/test-integration.yml

```
@@ -0,0 +1,77 @@
+name: Integration
+on:
+  pull_request:
+  push:
+    paths:
+      - '**.go'
+      - '**/go.mod'
+      - '**/go.sum'
+      - '**/testdata/**'
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  pre-test:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v3.4.0
+      - name: Finding files and store to output
+        id: set-matrix
+        run: echo "matrix=$({ cd integration && find . -type d ! -name testdata -maxdepth 1 -print; } | tail -n +2 | cut -c 3- | jq -R . | jq -cs .)" >> $GITHUB_OUTPUT
+
+  integration:
+    name: test ${{ matrix.test-path }}
+    runs-on: ubuntu-latest
+    needs: pre-test
+    if: fromJSON(needs.pre-test.outputs.matrix)[0] != null
+    continue-on-error: false
+    strategy:
+      fail-fast: true
+      matrix:
+        test-path: ${{fromJson(needs.pre-test.outputs.matrix)}}
+    steps:
+      - uses: actions/checkout@v3.4.0
+      - uses: technote-space/get-diff-action@v6.1.2
+        with:
+          PATTERNS: |
+            ${{ matrix.test-path }}/**
+            !${{ matrix.test-path }}/README.md
+            integration/go.mod
+            integration/go.sum
+            integration/${{ matrix.test-path }}/**
+
+      - uses: actions/cache@v3
+        if: env.GIT_DIFF
+        with:
+          path: |
+            ~/.cache/go-build
+            ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - uses: actions/setup-go@v4
+        if: env.GIT_DIFF
+        with:
+          go-version: '1.21'
+      - name: Run Integration Tests
+        if: env.GIT_DIFF
+        run: GOSUMDB=off go test -v -timeout 120m ./integration/${{ matrix.test-path }}
```

We should use something like this which was recently applied to the CLI repo, to make sure tests run with the features introduced in Go `v1.21`:

```suggestion
      - uses: actions/setup-go@v4
        if: env.GIT_DIFF
        with:
          go-version: 'stable'

      - name: Run Integration Tests
        if: env.GIT_DIFF
        env:
          GOTOOLCHAIN: local+path
          GOSUMDB: off
        run: go test -v -timeout 120m ./integration/${{ matrix.test-path }}
```

In short, these changes make sure that a recent Go version is installed to avoid trying to download a higher Go toolchain to run the tests. Toolchains can't be downloaded because `GOSUMDB=off`. Reason for that is that the CLI repo is too big and makes Go's sumdb timeout.